### PR TITLE
Chore: Enable builtin pre-commit checks

### PR DESCRIPTION
**Pull request by connortann**
Enables a selection of builtin pre-commit checks, which are fast to run and prevent some common issues.

Seeing as we have already enabled pre-commit, this seems like a reasonably unintrusive ci change.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,3 +3,10 @@ repos:
   rev: v0.0.265
   hooks:
   - id: ruff
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v2.3.0
+  hooks:
+  - id: check-added-large-files
+  - id: mixed-line-ending
```
Comment (thatlittleboy) [inline .pre-commit-config.yaml]: 1. Could you run `pre-commit autoupdate` to update the versions? I think pre-commit-hooks is currently at v4.4.0.
2. I propose adding `check-merge-conflict` and `end-of-file-fixer` as well.

**Review by thatlittleboy — CHANGES_REQUESTED** [review]
Always up for enforcing stringent ci/code quality checks :)

File: .pre-commit-config.yaml (hunk 1 above)
Comment (connortann) [reply .pre-commit-config.yaml]: Good suggestions, done. The end-of-file fixer adds a fair number of fixes as part of this PR.

**Comment by connortann** [comment]
I note that the codecov patch target is missed by this PR. I feel comfortable in this instance, as the changes are purely are made by the automatic `end-of-file-fixer` check and have no functional consequences. However, it does highlight the gaps in the test suite that we can hope to close over time.

**Review by thatlittleboy — APPROVED** [review]
trump-showing-lgtm

**Pull request merged**